File: backend/app/models/Pembelian.go
package models

import "time"

type Pembelian struct {
	ID               int       `json:"id"`
	NomorPembelian   string    `json:"nomor_pembelian"`
	SupplierID       int       `json:"supplier_id"`
	TanggalPembelian time.Time `json:"tanggal_pembelian"`
	Subtotal         float64   `json:"subtotal"`
	Diskon           float64   `json:"diskon"`
	Pajak            float64   `json:"pajak"`
	Total            float64   `json:"total"`
	Status           string    `json:"status"`
	Catatan          string    `json:"catatan"`
	CreatedBy        string    `json:"created_by"`
	CreatedAt        time.Time `json:"created_at"`
	UpdatedAt        time.Time `json:"updated_at"`
}

File: src/types/pembelian.ts
export type StatusPembelian = "draft" | "diterima" | "dibatalkan";

export interface PembelianDetail {
  id: string;
  pembelianId: string;
  barangId: string;
  namaBarang: string;
  batchId?: string;
  batchNumber?: string;
  tanggalExpired?: string;
  satuanId?: string;
  jumlah: number;
  hargaBeli: number;
  diskonPersen: number;
  diskonNominal: number;
  subtotal: number;
  hargaPokok: number;
}

export interface Pembelian {
  id: string;
  cabangId: string;
  nomorFaktur: string;
  nomorInternal: string;
  supplierId: string;
  namaSupplier: string;
  suratPesananId?: string;
  tanggalFaktur: string;
  tanggalJatuhTempo?: string;
  subtotal: number;
  diskonTotal: number;
  pajakTotal: number;
  grandTotal: number;
  status: StatusPembelian;
  catatan?: string;
  createdBy?: string;
  createdAt: string;
  updatedAt: string;
  details: PembelianDetail[];
}
